### ISSUE TITLE
Refactor agent and rename

### ISSUE THREAD

**Issue by alvarolobato** [elastic/apm-agent-rum-js#108]
We need to refactor the packaging in the agent to:
github repo: apm-agent-rum-js
npm: `@elastic/apm-rum`

- [x] Change the name of this repo
- [x] update the docs `conf.yml` to fetch the new repo. Even though Github automatically sets a redirect for the old url, we should update it to be safe. https://github.com/elastic/docs/pull/676
- [x] rename the bundle to `elastic-apm-rum` https://github.com/elastic/apm-agent-rum-js/pull/202
- [x] Publish the module to npm under the new name
- [x] Build docs to new `rum-js` URL: https://github.com/elastic/docs/pull/958
- [x] Redirect all docs to the new url https://github.com/elastic/website-www.elastic.co/issues/3674
- [x] Remove `js-base` form the code base (docs: https://github.com/elastic/apm-agent-rum-js/pull/378)
- [x] Fix Kibana instructions https://github.com/elastic/kibana/pull/32903 https://github.com/elastic/kibana/pull/32902 https://github.com/elastic/kibana/pull/32901 https://github.com/elastic/kibana/pull/32896
- [x] Update instructions in the solutions page https://github.com/elastic/content/issues/2972
- [x] Add deprecation warning to the old npm package

Targeting version 4.

Postponed:
- [x] Fix Kibana agent key https://github.com/elastic/kibana/issues/32643 (planned for 7.1)

**Comment by jahtalab** [comment]
@bmorelli25 , Is there a way for us to change the [url prefix](https://github.com/elastic/docs/blob/master/conf.yaml#L918) for the docs while still be able to redirect old prefix to the new one.

**Comment by bmorelli25** [comment]
So we're looking at changing the repo from `js-base` to `rum-js`? We can swap the prefix in the doc build and ask the web team to set up a redirect from: 
`https://www.elastic.co/guide/en/apm/agent/js-base/*` to `https://www.elastic.co/guide/en/apm/agent/rum-js/*`

Timing would be key to making this all work. When do you anticipate making this change?

**Comment by jahtalab** [comment]
I'm planning to do the required changes in the following steps:

* Publish the module to npm under the new name (it's currently being done)
* Change the name of this repo, we need to update the docs `conf.yml` to fetch the new repo. Even though Github automatically sets a redirect for the old url, we should update it to be safe.
* Change the docs prefix from `js-base` to `rum-js` and configuring the correct redirects. This step can be done at any time since all the links should work as they are, but we should do it once the previous steps are done.
* Update the links for our docs and our repo in Kibana instructions, we need to consider the next Kibana release.

**Comment by bmorelli25** [comment]
Great. That order is perfect!

> Change the docs prefix from `js-base` to `rum-js` and configuring the correct redirects. This step can be done at any time since all the links should work as they are, but we should do it once the previous steps are done.

A few days before we do this step ^ is when I'll open the redirect PR so the web team can have time to get everything ready. Ideally this prefix change and the redirects can be activated at the same time (or as close together as possible)

**Comment by jahtalab** [comment]
@bmorelli25 , We're ready to get started with setting up the new prefix and the required redirects, how should we proceed?

**Comment by bmorelli25** [comment]
Nice. I'll talk to the web team and see what they need from us.

**Comment by jahtalab** [comment]
@bmorelli25 , Is there any update on the docs redirect setup? Is there anything we can do to move it forward?

**Comment by bmorelli25** [comment]
Thanks for the bump on this @jahtalab. I never followed up on the issue, sorry about that. Looks like we still need some feedback from Nik before we can push this over. Unfortunately he's on vacation until July 8th.

**Comment by bmorelli25** [comment]
I'm opened a bunch of PRs to make the move to `rum-js`. Details are available here: https://github.com/elastic/docs/pull/958. If y'all can take a look and give a review to all of the PRs linked, that would be 👍

**Comment by jahtalab** [comment]
Thanks @bmorelli25 👍

**Comment by bmorelli25** [comment]
We're still waiting for Nik to get back from vacation. No one is sure what to do with the redirects, so we need him to move forward.

**Comment by bmorelli25** [comment]
Update: After a very, very long delay, RUM documentation now builds to `.../rum-js/*`. Redirects have been requested in the PR listed above.

**Comment by jahtalab** [comment]
Thanks for the update @bmorelli25 !
